File: example_services/scala/finatra/example/example-persistence/src/main/scala/com/me/example/persistence/InMemoryMessageDao.scala
package com.me.example.persistence

import java.util.concurrent.ConcurrentHashMap
import com.twitter.util.Future
import com.twitter.inject.Logging

object InMemoryMessageDao {
  val model: ConcurrentHashMap[String, String] = new ConcurrentHashMap[String, String]
}

class InMemoryMessageDao
  extends MessageDao
  with Logging {
  import InMemoryMessageDao._

  def get(key: String): Future[Option[String]] = Future {
    info("Using In Memory Dao for GET!!!")
    Option(model.get(key))
  }

  def set(key: String, value: String): Future[Unit] = Future {
    info("Using In Memory Dao for SET!!!")
    model.put(key, value)
  }

  def remove(key: String): Future[Unit] = Future {
    model.remove(key)
  }

  def update(key: String, value: String): Future[Unit] = Future {
    model.put(key, value)
  }
}

File: example_services/java/finatra/example/example-persistence/src/main/java/com/me/example/persistence/InMemoryMessageDao.java
package com.me.example.persistence;

import javax.inject.Singleton;
import java.util.Optional;
import java.util.concurrent.ConcurrentHashMap;

@Singleton
public class InMemoryMessageDao implements MessageDao {
    ConcurrentHashMap<String, String> model = new ConcurrentHashMap<>();

    @Override
    public Optional<String> get(String key) {
        return Optional.ofNullable(model.get(key));
    }

    @Override
    public void set(String key, String value) {
        model.put(key, value);
    }

    @Override
    public void remove(String key) {
        model.remove(key);
    }

    @Override
    public void update(String key, String value) {
        model.put(key, value);
    }
}
